modules/consensus/setdb.go

```diff
@@ -0,0 +1,226 @@
+package consensus
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/boltdb/bolt"
+)
+
+var meta = persist.Metadata{
+	Version: "0.1",
+	Header:  "Consensus Set Backing Database",
+}
+
+var (
+	errBadSetInsert = errors.New("attempting to add an already existing item to the consensus set")
+	errNilBucket    = errors.New("using a bucket that does not exist")
+	errNilItem      = errors.New("Requested item does not exist")
+)
+
+// setDB is a wrapper around the persist bolt db which backs the
+// consensus set
+type setDB struct {
+	*persist.BoltDatabase
+}
+
+// processedBlock is a copy/rename of blockNode, with the pointers to
+// other blockNodes replaced with block ID's, and all the fields
+// exported, so that a block node can be marshalled
+type processedBlock struct {
+	Block    types.Block
+	Parent   types.BlockID
+	Children []types.BlockID
+
+	Height      types.BlockHeight
+	Depth       types.Target
+	ChildTarget types.Target
+
+	DiffsGenerated            bool
+	SiacoinOutputDiffs        []modules.SiacoinOutputDiff
+	FileContractDiffs         []modules.FileContractDiff
+	SiafundOutputDiffs        []modules.SiafundOutputDiff
+	DelayedSiacoinOutputDiffs []modules.DelayedSiacoinOutputDiff
+	SiafundPoolDiffs          []modules.SiafundPoolDiff
+
+	ConsensusSetHash crypto.Hash
+}
```

Comment:
I know we are still in the transition period, but eventually this struct should fully replace `blockNode`